### PR TITLE
[CDAP-16630] Checking if source table definition instanceof StandardTableDefinition before casting

### PR DESCRIPTION
**Pull request by vravish**

### CODE REVIEW

**Review by chtyim — APPROVED** [review]

File: src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java [hunk 1]
```diff
@@ -392,10 +392,12 @@ private void validatePartitionProperties(FailureCollector collector) {
     if (sourceTable == null) {
       return;
     }
-    TimePartitioning timePartitioning = ((StandardTableDefinition) (sourceTable).getDefinition())
-      .getTimePartitioning();
-    if (timePartitioning == null) {
-      return;
+    if (sourceTable.getDefinition() instanceof StandardTableDefinition) {
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java:395]: can `sourceTable.getDefinition()` be instance of some other class? If so should we add a warn message saying that not supported or something similar?

Comment (vravish) [reply src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java:395]: The other class is autogenerated by BQ, and is only the definition if table's data comes from an external (non-BQ) source, e.g. Google Sheets.

Comment (CuriousVini) [reply src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java:395]: I see.

**Review by CuriousVini — APPROVED** [review]

**Comment by albertshau** [comment]
@vravish be sure to cherrypick to release branches as well

File: src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java [hunk 2]
```diff
@@ -392,10 +392,12 @@ private void validatePartitionProperties(FailureCollector collector) {
...
+      TimePartitioning timePartitioning = ((StandardTableDefinition) sourceTable.getDefinition())
+          .getTimePartitioning();
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java:397]: indentation is off (should be 2 spaces). Please make sure your IDE has the right settings.

**Pull request merged**